Source/WebCore/PAL/pal/PALSwift/UnsafeOverlays.swift

```diff
@@ -30,12 +30,92 @@ import CryptoKit
 
 import PALSwift
 
+enum UnsafeErrors: Error {
+    case invalidLength
+    case emptySpan
+}
+
 extension HashFunction {
     mutating func update(data: SpanConstUInt8) {
-        self.update(
-            bufferPointer: UnsafeRawBufferPointer(
-                start: data.__dataUnsafe(), count: data.size()))
+        if data.empty() {
+            self.update(data: Zero.zeroData)
+        } else {
+            self.update(
+                bufferPointer: UnsafeRawBufferPointer(
+                    start: data.__dataUnsafe(), count: data.size()))
+        }
+    }
+}
+
+// Ugly but needed.
+// all __dataUnsafe() calls here return nil if input WTF::Vector is zero length.
+// CommonCrypto APIs handled nil pointers internally. Swift APIs crash.
+// To maintain backward compatibility, a zero array is needed to go past Swift Validation
+// and let CyptoKit replicate behavior.
+private struct Zero {
+    public static var zeroData = Data([UInt8]())
+}
+
+extension ContiguousBytes {
+    public func createVectorUInt8() -> VectorUInt8 {
+        return self.withUnsafeBytes { buf in
+            var result = VectorUInt8(buf.count)
+            buf.copyBytes(to: UnsafeMutableRawBufferPointer(start: UnsafeMutableRawPointer(mutating: result.__dataUnsafe()), count: result.size()), count: result.size())
+            return result
+        }
+    }
+}
+
+extension Data {
+    public static func fromSpan(spanNoCopy: SpanConstUInt8) -> Data {
```

Comment:
We should make this fileprivate because it's not a safe concept to export. Maybe all extensions should be so?

Maybe also name this unsafeTemporaryDataFromSpan, to remind us not to return it. (We decided not to do `Data.temporaryDataFromSpan { data in ... }` because you would end up nested a few levels deep, which is not super readable.)